package-res/resources/web/pentaho/type/_type.js

```diff
@@ -0,0 +1,1362 @@
+      // undefined -> inherit
+      // null -> clear
+      // "" -> null conversion
+
+      _view: null, // {value: any, promise: Promise.<Class.<View>>}
+
+     /**
+      * Gets or sets the default view for instances of this type.
+      *
+      * When a string,
+      * it is the id of the view's AMD module.
+      * If the string starts with `/` or `xyz:`, or ends with `.js`,
+      * the id is considered to be absolute,
+      * otherwise,
+      * it is relative to this type's id folder, and converted to an absolute id.
+      *
+      * Setting to `undefined` causes the view to be inherited from the ancestor type,
+      * except for the root type, `Instance.type` (which has no ancestor), where the attribute is `null`.
+      *
+      * Setting to a _falsy_ value (like `null` or an empty string),
+      * clears the value of the attribute and sets it to `null`, ignoring any inherited value.
+      *
+      * When a function,
+      * it is the class or factory of the view.
+      *
+      * @see pentaho.type.Type#viewClass
+      *
+      * @type {string | function}
+
+      * @throws {pentaho.lang.ArgumentInvalidTypeError} When the set value is not
+      * a string, a function or {@link Nully}.
+      */
+      get view() {
+        return this._view && this._view.value;
+      },
+
+      set view(value) {
+        if(value === undefined) {
+          this._resetView();
+        } else if(!value) { // null || ""
+          this._view = null;
+        } else  if(typeof value === "string") {
+          value = this._buildRelativeId(value);
+          if(!this._view || this._view.value !== value) {
+            this._view = {value: value, promise: null};
+          }
+        } else if(typeof value === "function") {
+          // Assume it is the View class itself, already fulfilled.
+          if(!this._view || this._view.value !== value) {
+            this._view = {value: value, promise: Promise.resolve(value)};
+          }
+        } else {
+          throw error.argInvalidType("view", ["nully", "string", "function"], typeof value);
+        }
+      },
+
+      _resetView: function() {
+        if(this !== _type) {
+          delete this._view;
+        }
+      },
+
+      /**
+       * Gets a promise for the default view class or factory, if any, or `null`.
+       *
+       * A default view exists if property {@link pentaho.type.Type#view}
+       * has a non-null value.
+       *
+       * @type Promise.<?function>
+       * @readOnly
+       * @see pentaho.type.Type#view
+       */
+      get viewClass() {
+        /*jshint laxbreak:true*/
+        var view = this._view;
+        return view
+            ? (view.promise || (view.promise = promiseUtil.require(view.value, localRequire)))
+            : Promise.resolve(null);
+      },
+      //endregion
+
+      /**
+       * Creates a subtype of this one.
+       *
+       * This method creates a subtype which does not have own instance or type constructors.
```

Comment:
Suggest:

This method creates a subtype which does not have own its instance or type constructors.